backend/app/api/routes/calendar_routes.py

```
@@ -0,0 +1,45 @@
+
+@router.get("/events")
+def get_calendar_events(year: int = datetime.now().year):
+    """ 
+    Fetch Google Calendar ICS feed and return JSON events for the given year.
+    Default = current year.
+    """
+    try:
+        res = requests.get(ICS_URL, timeout=10)
+        res.raise_for_status()
+    except requests.RequestException as e:
+        raise HTTPException(status_code=502, detail=f"Failed to fetch calendar feed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)